File: CursoPOOUber/Csharp/UberPool.cs
using CarNS;
using AccountNS;

namespace CarTypes.UberPool 
{

    //Definition of the "UberPool" class, which inherits from "Car" class.
    public class UberPool : Car 
    {

        //Attributes
        public string Brand { get; set; }
        public string Model { get; set; }

        public UberPool(string License, Account Driver, string brand, string model) : base(License, Driver) 
        {
            this.Brand = brand;
            this.Model = model;
            SetPassengers(4);
        }

        public override void PrintDataCar()
        {
            base.PrintDataCar();
            Console.WriteLine("Car Brand: " + Brand + " UberPool model: " + Model);
        }
    }
}

File: CursoPOOUber/PHP/uberPool.php
<?
require_once("car.php");
class UberPool extends Car {
    public $brand;
    public $model;

    public function __construct($license, $driver, $brand, $model){
        parent::__construct($license,$driver);
        $this->brand = $brand;
        $this->model = $model;
    }

    public function setPassenger($passenger) {
        if ($passenger == 4) {
            $this->passenger = $passenger;
        } else {
            echo "This type of Uber is only for 4 passengers.";
        }
    }

    public function printDataCar() {
        echo "Car brand: $this->brand; <br>
                UberPool model: {$this->model} <br>";
    }
}
?>